docs/samples/deployments/spark/presidio_anonymize_blobs.py

```diff
@@ -0,0 +1,153 @@
+
+# COMMAND ----------
+
+storage_account_name = dbutils.widgets.get("storage_account_name")  # noqa F821
+storage_container_name = dbutils.widgets.get("storage_container_name")  # noqa F821
+storage_account_access_key = dbutils.widgets.get(
+    "storage_account_access_key"
+)  # noqa F821
+
+
+blob_service_client = BlobServiceClient(
+    account_url="https://" + storage_account_name + ".blob.core.windows.net/",
+    credential=storage_account_access_key,
+)
+container_client = blob_service_client.get_container_client(storage_container_name)
+
+blob_names = container_client.list_blobs(
+    name_starts_with=dbutils.widgets.get("storage_input_folder") + "/"
+)
+blobs = []
+for blob in blob_names:
+    blobs.append(
+        "wasbs://"
+        + storage_container_name
+        + "@"
+        + storage_account_name
+        + ".blob.core.windows.net/"
+        + blob.name
+    )
+
+spark.conf.set(
+    "fs.azure.account.key." + storage_account_name + ".blob.core.windows.net",
+    storage_account_access_key,
+)
+
+input_df = spark.read.text(blobs).withColumn("filename", input_file_name())
+display(input_df)
+
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC # Anonymize Text using Presidio
+
+# COMMAND ----------
+
+
+def anonymize_text(text: str) -> str:
+    analyzer = AnalyzerEngine()
+    anonymizer = AnonymizerEngine()
+    analyzer_results = analyzer.analyze(text=text, language="en")
+    anonymized_results = anonymizer.anonymize(
+        text=text,
+        analyzer_results=analyzer_results,
+        anonymizers_config={
+            "DEFAULT": AnonymizerConfig("replace", {"new_value": "<ANONYMIZED>"})
+        },
+    )
+    return anonymized_results
+
+
+def anonymize_series(s: pd.Series) -> pd.Series:
+    return s.apply(anonymize_text)
+
+
+anonymize = pandas_udf(anonymize_series, returnType=StringType())
+
+anonymized_df = input_df.withColumn("anonymized_text", anonymize(col("value")))
+display(anonymized_df)
+
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC # Write the Anonymized content back to Azure Blob storage
+
+# COMMAND ----------
+
+anonymized_df = anonymized_df.withColumn(
+    "filename",
+    regexp_replace(
+        "filename",
+        "^.*(/" + dbutils.widgets.get("storage_input_folder") + "/)",
+        dbutils.widgets.get("storage_output_folder") + "/",
+    ),
+)
+
+
+def upload_to_blob(text, file_name):
+    blob_client = blob_service_client.get_blob_client(
+        container=storage_container_name, blob=file_name
+    )
+    blob_client.upload_blob(text)
+    return "SAVED"
+
+
+def upload_series(s1: pd.Series, s2: pd.Series) -> pd.Series:
+    res = []
+    for index, s1_item in s1.items():
+        s2_item = s2[index]
+        res.append(upload_to_blob(s1_item, s2_item))
+    return pd.Series(res)
+
+
+save_udf = pandas_udf(upload_series, returnType=StringType())
+
+out_df = anonymized_df.withColumn(
+    "processed", save_udf(col("anonymized_text"), col("filename"))
+)
+out_df.collect()
```

Comment:
I can see the intention here is to store the files back to the output directory and preserve the filenames. This one is tricky to do with native Spark. In order to avoid creating Blob Client and writing another Pandas UDF, what if we tried something like this;
- Group the final dataframe by filenames e.g. `anonymized_df.groupBy(col("filename"))`
- Then for each group, use an aggregate function to consolidate all text
- The above will output a condensed dataframe which can be simply looped over and stored to blob storage with vanilla Python filesystem API. You might have to remove the `wasbs:///` to use Python FS API but that can be solved with [mounting the blob storage to DBFS](https://docs.databricks.com/data/data-sources/azure/azure-storage.html#mount-azure-blob-storage-containers-to-dbfs).